robotidy/transformers/SmartSortKeywords.py

```diff
@@ -0,0 +1,81 @@
+from robot.api.parsing import ModelTransformer, EmptyLine
+from robot.parsing.model.blocks import Keyword
+
+
+class SmartSortKeywords(ModelTransformer):
+    """
+    Sort keywords in *** Keywords *** section.
+
+    By default sortin is case insensitve, but keywords with leading underscore go to the bottom. Other underscores are treated as spaces.
+    Empty lines (or lack of them) between keywords is preserved.
+
+    Following code::
+    *** Keywords ***
+    _my secrete keyword
+        Kw2
+
+    My Keyword
+        Kw1
+
+
+    my_another_cool_keyword
+    my another keyword
+        Kw3
+
+    Will be transformed to::
+    *** Keywords ***
+    my_another_cool_keyword
+
+    my another keyword
+        Kw3
+
+
+    My Keyword
+        Kw1
+    _my secrete keyword
+        Kw2
+
+    Default behaviour could be changed using following parameters: ``case_insensitive``, ``ignore_leading_underscore`` and ``ignore_other_underscore``
+    """
+
+    def __init__(self, case_insensitive=True, ignore_leading_underscore=False, ignore_other_underscore=True):
+        self.ci = case_insensitive
+        self.ilu = ignore_leading_underscore
+        self.iou = ignore_other_underscore
+
+    def visit_KeywordSection(self, node):  # noqa
+        if not node.body:
+            return node
+        empty_lines = self.pop_empty_lines(node)
+        node.body.sort(key=self.sort_function)
+        self.append_empty_lines(node, empty_lines)
+        return node
+
+    @staticmethod
+    def pop_empty_lines(node):
+        all_empty = []
+        for kw in node.body:
+            kw_empty = []
+            for index in range(len(kw.body) - 1, -1, -1):
+                if not isinstance(kw.body[index], EmptyLine):
+                    break
+                kw_empty.insert(0, kw.body.pop(index))
```

Comment:
If I understand it correctly: it does start from the end of keyword and pop every empty line into kw_empty. If there is something else than empty line - it stops. 

So we could write it that way (not tested):
```suggestion
            while kw.body and isistance(kw.body[-1], EmptyLine):
                kw_empty.insert(0, kw.body.pop())
```

Reply:
Still empty line wll be bit messed up after that transformer - that's why it would be good to either ensure it runs after new line normalization (for that add name of the transformer at the beginning of the TRANSFORMERS frozenset) or we would need to wait until I improve how the robotidy transform the code (I have some plans to deal with such problems). For now we can go with first option

Reply:
I've intentionally did not deal with empty/new lines as there is already transformer for this.